### DISCUSSION TITLE
Initialisation using VTK format files

### DISCUSSION THREAD

**Discussion by zengyuxin0807** [idaholab/moose#28054] [Q&A Meshing] [answered]
Dear Moose Experts,

I have a VTK storage file with information on stresses, strains, etc. that I want to use as an initial condition input. I see in the moose instructions that many kinds of files can be read through the mesh board, but I didn't find an example of a relevant reading of a VTK.

How do I read my vtk into moose as an initialisation condition, and if I can read the vtk what does it look like in terms of parameter naming and data storage format. If I can't read the vtk, what format do I need to convert it to and how do I do it?

I would greatly appreciate any insights or suggestions from the community to help resolve this issue.

Thank you very much for your time and consideration.

**Comment by milljm** [comment]
We have a VTK file writer/reader object available. I am not very sure how to interface with it. You might be able to figure it out on your own if you look at the following test spec files which make use of the library:

```bash
moose/test/tests/functions/image_function/test
```
And then the input files associated with those tests.


I believe @lindsayad can describe this in greater detail.

**Reply by GiudGiud** [reply, marked as answer]
Hello

I dont think the VTK file can be used for anything outside of the mesh in MOOSE. you cannot initialize fields from it
However, using meshio, a Python library, you can convert your VTK file to Exodus, and if this works well, then you can use Exodus restart (one of the two ways, explained here https://mooseframework.inl.gov/application_usage/restart_recover.html)

**Reply by zengyuxin0807** [reply]
Thank you very much for the solution ideas, this has helped me a lot, I will try them out!

**Reply by lindsayad** [reply]
@GiudGiud is correct that we don't support reading solution information from VTK. Probably not useful to you, but we do support writing solution information to VTK
